### PR TITLE
Update readme to install skipper

### PR DESCRIPTION
**Pull request by kostspielig**
To install skipper without having the source with the .mod file the flag should not be on

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ Skipper is 'go get' compatible. If needed, create a Go workspace first:
 
 Get the Skipper packages:
 
-    GO111MODULE=on go get github.com/zalando/skipper/...
+    GO111MODULE=auto go get github.com/zalando/skipper/...
```
Comment (aryszka) [inline readme.md]: isn't auto the default?

**Comment by aryszka** [comment]
this PR is a very recent change: https://github.com/zalando/skipper/pull/872/files

can you discuss with @szuecs the problem and the best approach here?

**Comment by aryszka** [comment]
i just tested this, and both without the flag and with 'on', and neither really works. I expect that the current working directory needs to be the skipper repo. We probably need to document this after testing that it works.

**Comment by kostspielig** [comment]
Both options seem to give errors then:
```
✓ maria ~/go $ GO111MODULE=on go get github.com/zalando/skipper/...
go: cannot find main module; see 'go help modules'
```
Without modules it installs but seems to use the incorrect version of the libraries
```
✓ maria ~/go $ go get -u github.com/zalando/skipper/...
# github.com/uber/jaeger-client-go
src/github.com/uber/jaeger-client-go/metrics.go:100:35: too many arguments in call to factory.Namespace
	have (string, nil)
	want (metrics.NSOptions)
```

Maybe the best option then is to clone the repo and download dependencies with GO111MODULE on then?

**Comment by szuecs** [comment]
@kostspielig maybe you can find out why go get breaks in case of `GO111MODULE=on`?

**Comment by szuecs** [comment]
https://github.com/zalando/skipper/issues/871#issuecomment-444402621 said that it worked. Maybe a Mac issue? 
Does it work if you clone the repository and use `GO111MODULE=on go get github.com/zalando/skipper/...` ?

**Comment by szuecs** [comment]
```
% GO111MODULE=on go get github.com/zalando/skipper
go: finding github.com/cenkalti/backoff v2.1.0+incompatible
go: downloading github.com/cenkalti/backoff v2.1.0+incompatible
% GO111MODULE=on go get github.com/zalando/skipper/...
# github.com/zalando/skipper/skptesting
ld: warning: text-based stub file /System/Library/Frameworks//CoreFoundation.framework/CoreFoundation.tbd and library file /System/Library/Frameworks//CoreFoundation.framework/CoreFoundation are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks//Security.framework/Security.tbd and library file /System/Library/Frameworks//Security.framework/Security are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks//IOKit.framework/Versions/A/IOKit.tbd and library file /System/Library/Frameworks//IOKit.framework/Versions/A/IOKit are out of sync. Falling back to library file for linking.
# github.com/zalando/skipper/cmd/skipper
ld: warning: text-based stub file /System/Library/Frameworks//CoreFoundation.framework/CoreFoundation.tbd and library file /System/Library/Frameworks//CoreFoundation.framework/CoreFoundation are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks//Security.framework/Security.tbd and library file /System/Library/Frameworks//Security.framework/Security are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks//IOKit.framework/Versions/A/IOKit.tbd and library file /System/Library/Frameworks//IOKit.framework/Versions/A/IOKit are out of sync. Falling back to library file for linking.
% uname -a
Darwin uho.local 17.7.0 Darwin Kernel Version 17.7.0: Fri Nov  2 20:43:16 PDT 2018; root:xnu-4570.71.17~1/RELEASE_X86_64 x86_64

% go env
GOARCH="amd64"
GOBIN="/Users/sszuecs/go/bin"
GOCACHE="/Users/sszuecs/Library/Caches/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/sszuecs/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/sszuecs/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/22/hffqd9zj5dx0b2b07z6c0b980000gq/T/go-build958763120=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

**Comment by szuecs** [comment]
Yes, please update the docs and link the "why".

**Comment by kostspielig** [comment]
@szuecs were you able to reproduce or it worked for you outside the repo?

**Comment by szuecs** [comment]
@kostspielig for me it worked as I showed on a mac in a blank directory which was not in $GOPATH.

**Comment by szuecs** [comment]
@kostspielig do you want to work on this or should we close it?

**Comment by szuecs** [comment]
There was no answer after 6 months so close

**Pull request closed without merging**